src/main/java/com/bookstore/repository/ShoppingCartRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.bookstore.repository;
+
+import com.bookstore.entity.ShoppingCart;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+    @EntityGraph(attributePaths = {"cartItems", "user", "cartItems.book"})
+    @Query("select c from ShoppingCart c where c.user.email = :email")
```

Comment:
why did you use custom query if it's enough to follow JPA method name conventions and JPA will create the method with the supposed query itself? fix the same issue in all places